Core/source/core/pgp-key.ts

```diff
@@ -349,32 +355,36 @@ export class PgpKey {
    * Get latest self-signature date, in utc millis.
    * This is used to figure out how recently was key updated, and if one key is newer than other.
    */
-  public static lastSig = async (key: OpenPGP.key.Key): Promise<number> => {
+  public static lastSig = async (key: Key): Promise<number> => {
+    // no longer works, need some alternate solution
+    // discussion is in progress: https://github.com/openpgpjs/openpgpjs/discussions/1491
     await key.getExpirationTime(); // will force all sigs to be verified
-    const allSignatures: OpenPGP.packet.Signature[] = [];
+    const allSignatures: SignaturePacket[] = [];
     for (const user of key.users) {
       allSignatures.push(...user.selfCertifications);
     }
-    for (const subKey of key.subKeys) {
+    for (const subKey of key.subkeys) {
       allSignatures.push(...subKey.bindingSignatures);
     }
-    allSignatures.sort((a, b) => b.created.getTime() - a.created.getTime());
+    allSignatures.sort((a, b) => (b.created ? b.created.getTime() : 0) - (a.created ? a.created.getTime() : 0));
     const newestSig = allSignatures.find(sig => sig.verified === true);
     if (newestSig) {
-      return newestSig.created.getTime();
+      return newestSig.created ? newestSig.created.getTime() : 0;
     }
     throw new Error('No valid signature found in key');
   }
```

Comment:
Highlighting to keep track

Reply:
@tomholub Please read carefully this one https://github.com/openpgpjs/openpgpjs/discussions/1491#discussioncomment-2149242
What do you think of special cases mentioned there?

Reply:
I've implemented new version of the `lastSig()`. Asked for review from openpgp maintainers.